src/Claim.ts

```diff
@@ -1,14 +1,76 @@
 /* tslint:disable:no-relative-imports */
-import * as bitcore from 'bitcore-lib'
 import * as crypto from 'crypto'
-import { canonize } from 'jsonld'
+import * as cuid from 'cuid'
+import * as jsonld from 'jsonld'
+import * as jsig from 'jsonld-signatures'
 
 import { IllegalArgumentException } from './Exceptions'
-import { Claim, ClaimAttributes, ClaimType, ClaimContext, isClaim } from './Interfaces'
+import { Claim, ClaimType, ClaimContext, DefaultClaimContext, claimTypeDefaults, isClaim } from './Interfaces'
+import { DataParser } from './util/DataParser'
 
-export const canonizeClaim = async (claim: Claim): Promise<string> => {
-  const contextualClaim = { ...ClaimContext, ...claim }
-  return canonize(contextualClaim)
+DataParser(jsonld)
```

Comment:
This has been bugging me for a while, but I think I may have found a solution.

Mutating an import could be dangerous, leading to odd and hard-to-find bugs. 

Please see this small example I wrote to showcase what's going on when we mutate an import: [node-import-mutation](https://github.com/lautarodragan/node-import-mutation/blob/master/README.md).

Even though it isn't documented anywhere (I could find), `jsonld.js` [exports a factory](https://github.com/digitalbazaar/jsonld.js/blob/1fae88fd856be2bdb63e3a9d71d54e848a1aafe5/lib/jsonld.js#L1065-L1099). It does so in a very odd manner, which allows us to use the dependency both as a factory and as a singleton, the dependency itself.

This means we can do the following:

```js
const JSONLD = require('jsonld')
const jsonld = JSONLD()

jsonld.someVar = { someVar: 42 }
console.log('jsonld.someVar', jsonld.someVar)
console.log('JSONLD.someVar', JSONLD.someVar)
console.log('jsonld.documentLoader', jsonld.documentLoader)
console.log('JSONLD.documentLoader', JSONLD.documentLoader)
```

This prints the following:

```
$ node index.js 
jsonld.someVar { someVar: 42 }
JSONLD.someVar undefined
jsonld.documentLoader async function(url) {
    const callback = arguments[1];
    return new Promise((resolve, reject) => {
      try {
        fn(url, (err, remoteDoc) => {
          if(typeof callback === 'function') {
            return _invokeCallback(callback, err, remoteDoc);
          } else if(err) {
            reject(err);
          } else {
            resolve(remoteDoc);
          }
        });
      } catch(e) {
        if(typeof callback === 'function') {
          return _invokeCallback(callback, e);
        }
        reject(e);
      }
    });
  }
JSONLD.documentLoader async function(...args) {
    const callback = args[args.length - 1];
    if(typeof callback === 'function') {
      args.pop();
    }

    let result;
    try {
      result = await fn.apply(null, args);
    } catch(e) {
      if(typeof callback === 'function') {
        return _invokeCallback(callback, e);
      }
      throw e;
    }

    if(typeof callback === 'function') {
      return _invokeCallback(callback, null, result);
    }

    return result;
  }
```

I haven't been able to test using the library as a factory and working on an instance of the API rather than the singleton, but if we can do that, it'd be ideal. 

That would allow us to keep Po.etJS 100% stateless.

Reply:
💯

Reply:
Nicely done, and great job on the research documentation!

Reply:
Lautaro, from our phone discussion, I do not understand how you want to pull jsonld/jsonld-signatures completely out of poet-js. This would greatly complicate `getClaimId`, `signClaim`, `isValidClaim`, `createClaim`. This is akin to removing protobuf, and any rsa signing method, hashing methods we have in the code.

I can see a `CreateSigner` object that pulls all of the necessary items, but I think it belongs insided poet-js not outside... and it should include jsonld.